### PR TITLE
Switch from setup.py to pyproject.toml

### PR DESCRIPTION
**Pull request by waylan**
This completely replaces `setup.py` with `pyproject.toml`.

@facelessuser and @mitya57 if either of you would like to be listed in the `maintainers` entry, I would be happy to add you. You can be listed with or without an email address. If you would also like your email address listed, provide it and I'll include it as well.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -2,3 +2,88 @@
 # Minimum requirements for the build system to execute.
 requires = ["setuptools>=36.6", "wheel"]
```
Comment (mitya57) [inline pyproject.toml]: I think you now need to bump minimum setuptools version to 61 (or even 61.2, as some recommend).

**Review by mitya57 — COMMENTED** [review]
I don't do much these days, so it's okay to not list me in maintainers.

**Comment by facelessuser** [comment]
Yeah, you can add me.

File: pyproject.toml (hunk 1 above)
Comment (waylan) [reply pyproject.toml]: Good call. Thanks.

**Comment by waylan** [comment]
@facelessuser I have added your name. If you can confirm I spelled it correctly, etc. then I will merge this.

**Pull request merged**